src/format.py

```diff
@@ -155,15 +156,30 @@ def updateDecoratedMatch(self):
         else:
             attributes = (0, 0, FormattedText.UNDERLINE_ATTRIBUTE)
 
+        decoratorText = self.getDecorator()
+
         self.decoratedMatch = FormattedText(
             FormattedText.getSequenceForAttributes(*attributes) +
-            self.getDecorator() + self.getMatch())
+            decoratorText + self.getMatch())
+
+        # because decorators add length to the line, when the decorator
+        # is removed, we need to print blank space (aka "erase") the
+        # part of the line that is stale. calculate how much this is based
+        # on the max length decorator.
+        self.endingClearText = FormattedText(
+            FormattedText.getSequenceForAttributes(
+                FormattedText.DEFAULT_COLOR_FOREGROUND,
+                FormattedText.DEFAULT_COLOR_BACKGROUND, 0) +
+            " " * (self.getMaxDecoratorLength() - len(decoratorText)))
```

Comment:
hrm, im not a huge fan of this since then we are starting to print blank spaces after the lines **regardless* of whether we just become selected or not. you can see this in all the test input which has a bunch of trailing whitespace :-/

can we instead just dirty the line completely in screen control and use that to handle the refresh?